RollbarNotifier/Sources/RollbarNotifier/RollbarPayloadFactory.m

```diff
@@ -262,64 +263,45 @@ -(RollbarServer *)buildRollbarServer {
-    
-    if (self->_osData) {
-        return self->_osData;
+- (NSDictionary<NSString *, id> *)buildOSData {
+    if (self.osData) {
+        return self.osData;
     }
-    
+
     NSBundle *mainBundle = [NSBundle mainBundle];
-    
-    NSString *version = nil;
-    if (self->_config.loggingOptions
-        && self->_config.loggingOptions.codeVersion
-        && self->_config.loggingOptions.codeVersion.length > 0) {
-        
-        version = self->_config.loggingOptions.codeVersion;
-    }
-    else {
-        version = [mainBundle objectForInfoDictionaryKey:(NSString*)kCFBundleVersionKey];
-    }
-    
+    NSString *codeVersion = [mainBundle objectForInfoDictionaryKey:(NSString *)kCFBundleVersionKey];
     NSString *shortVersion = [mainBundle objectForInfoDictionaryKey:@"CFBundleShortVersionString"];
     NSString *bundleName = [mainBundle objectForInfoDictionaryKey:(NSString *)kCFBundleNameKey];
     NSString *bundleIdentifier = [mainBundle objectForInfoDictionaryKey:(NSString *)kCFBundleIdentifierKey];
-    
+
     struct utsname systemInfo;
     uname(&systemInfo);
-    NSString *deviceCode = [NSString stringWithCString:systemInfo.machine
-                                              encoding:NSUTF8StringEncoding];
-    
+    NSString *deviceCode = [NSString stringWithCString:systemInfo.machine encoding:NSUTF8StringEncoding];
+
 #if TARGET_OS_IOS | TARGET_OS_TV | TARGET_OS_MACCATALYST
-    self->_osData = @{
+    self.osData = @{
         @"os": @"iOS",
         @"os_version": [[UIDevice currentDevice] systemVersion],
         @"device_code": deviceCode,
-        @"code_version": version ? version : @"",
-        @"short_version": shortVersion ? shortVersion : @"",
-        @"bundle_identifier": bundleIdentifier ? bundleIdentifier : @"",
-        @"app_name": bundleName ? bundleName : @""
+        @"code_version": codeVersion ?: @"",
+        @"short_version": shortVersion ?: @"",
+        @"bundle_identifier": bundleIdentifier ?: @"",
+        @"app_name": bundleName ?: @""
     }.mutableCopy;
 #else
-    NSOperatingSystemVersion osVer = [[NSProcessInfo processInfo] operatingSystemVersion];
-    self->_osData = @{
+    NSOperatingSystemVersion v = [[NSProcessInfo processInfo] operatingSystemVersion];
+    self.osData = @{
         @"os": @"macOS",
-        @"os_version": [NSString stringWithFormat:@" %tu.%tu.%tu",
-                        osVer.majorVersion,
-                        osVer.minorVersion,
-                        osVer.patchVersion
-        ],
+        @"os_version": [NSString stringWithFormat:@" %tu.%tu.%tu", v.majorVersion, v.minorVersion, v.patchVersion],
         @"device_code": deviceCode,
-        @"code_version": version ? version : @"",
-        @"short_version": shortVersion ? shortVersion : @"",
-        @"bundle_identifier": bundleIdentifier ? bundleIdentifier : @"",
-        @"app_name": bundleName ? bundleName : [[NSProcessInfo processInfo] processName]
+        @"code_version": version ?: @"",
```

Comment:
Exactly. It's called the [Elvis operator](https://en.wikipedia.org/wiki/Elvis_operator), and since what's being compared is a pointer to a value, if the value doesn't exist, the pointer points to 0, and 0 is false. If the value exists, the pointer points to "something that is not 0", and that means true.